src/vehicle.cpp

```diff
@@ -3494,6 +3456,24 @@ void vehicle::idle(bool on_map) {
             }
 
             if( on_map ) {
+                static std::map<int, std::string> opts = {
```

Comment:
That's an awful lot of allocation. Could you turn this into an array of ints and pick which sound string to construct so that only one call to _() is needed?